libafl/src/corpus/mod.rs

```diff
@@ -148,6 +148,18 @@ pub trait Corpus: UsesInput + Serialize + for<'de> Deserialize<'de> {
     }
 }
 
+/// Trait for types which track the current corpus index
+pub trait HasCurrentCorpusIdx {
+    /// Set the current corpus index; we have started processing this corpus entry
+    fn set_corpus_idx(&mut self, idx: CorpusId) -> Result<(), Error>;
```

Comment:
I intentionally chose not to do this because it allows for a callback to be invoked when the corpus index is set (e.g. if there's any metadata or smth).